internal/cephfs/snapshot.go

```diff
@@ -168,6 +177,11 @@ func protectSnapshot(ctx context.Context, volOptions *volumeOptions, cr *util.Cr
 }
 
 func unprotectSnapshot(ctx context.Context, volOptions *volumeOptions, cr *util.Credentials, snapID, volID volumeID) error {
+	// If "snapshot-autoprotect" feature is present, The UnprotectSnapshot
+	// call should be treated as a no-op.
+	if checkSubvolumeHasFeature(autoProtect, volOptions.Features) {
```

Comment:
`volumeOptions` missing Features when invoked from `DeleteSnapshot`.

`volOpt` in DeleteSnapshot is populated in function `newSnapshotOptionsFromID` which does not invoke parent vol infor to read features and populate the same.

Reply:
@ShyamsundarR as subvolume and snapshots are independent is it possible to get subvolume info if subvolume is already deleted?

Reply:
Subvolume when deleted using `--retain-snapshots` retains the subvolume in a retained state, and `subvolume info` is still valid and will return the subvolume `type`, `features`, and `state` to the caller.